src/main/kotlin/com/ampnet/web3provider/repository/ProviderRedisRepository.kt

```diff
@@ -0,0 +1,9 @@
+package com.ampnet.web3provider.repository
+
+import java.math.BigInteger
+
+interface ProviderRedisRepository {
```

Comment:
`Provider` part sounds a bit weird to me here, as if this should have been called `RedisRepositoryProvider`, but from the methods I don't think that is the case... Maybe try calling it `BalanceProviderRedisRepository` or something like that to make it a bit clearer? Maybe even just `BalanceRepository` and then the implementation can be called `RedisBalanceRepository` or `BalanceRedisRepository`.

Reply:
Ok, sure, I''l think of a more suitable naming.